File: app/transports/openapi/bindings/info.go
package bindings

import (
	"context"

	"github.com/Southclaws/fault"
	"github.com/Southclaws/fault/fctx"

	"github.com/Southclaws/storyden/app/resources/settings"
	"github.com/Southclaws/storyden/app/services/onboarding"
	"github.com/Southclaws/storyden/internal/openapi"
)

type Info struct {
	sr settings.Repository
	os onboarding.Service
}

func NewInfo(sr settings.Repository, os onboarding.Service) Info {
	return Info{
		sr: sr,
		os: os,
	}
}

func (i Info) GetInfo(ctx context.Context, request openapi.GetInfoRequestObject) (openapi.GetInfoResponseObject, error) {
	settings, err := i.sr.Get(ctx)
	if err != nil {
		return nil, fault.Wrap(err, fctx.With(ctx))
	}

	status, err := i.os.GetOnboardingStatus(ctx)
	if err != nil {
		return nil, fault.Wrap(err, fctx.With(ctx))
	}

	return openapi.GetInfo200JSONResponse{
		GetInfoOKJSONResponse: openapi.GetInfoOKJSONResponse{
			Title:            settings.Title.Get(),
			Description:      settings.Description.Get(),
			AccentColour:     settings.AccentColour.Get(),
			OnboardingStatus: openapi.OnboardingStatus(status.String()),
		},
	}, nil
}

File: web/src/utils/info.ts
import { Info } from "@/api/openapi-schema";
import { getInfo as getInfoAPI } from "@/api/openapi-server/misc";

import { FALLBACK_COLOUR } from "./colour";

export async function getInfo(): Promise<Info> {
  try {
    const { data } = await getInfoAPI();
    return data;
  } catch (e) {
    console.error(e);
    return {
      title: "Storyden",
      description: "A forum for the modern age.",
      accent_colour: FALLBACK_COLOUR,
      onboarding_status: "complete",
    };
  }
}
